pennylane/interfaces/torch.py

```diff
@@ -16,13 +16,69 @@
 compatible quantum nodes.
 """
 # pylint: disable=redefined-outer-name,arguments-differ
+from collections import Iterable
 import inspect
 from functools import partial
+import numbers
 
 import numpy as np
 import torch
+from torch.autograd.function import once_differentiable
 
-from pennylane.utils import unflatten
+
+def _unflatten_torch(flat, model):
+    """Restores an arbitrary nested structure to a flattened Torch tensor.
+
+    See also :func:`_flatten`.
+
+    Args:
+        flat (torch.Tensor): 1D array of items
+        model (array, Iterable, Number): model nested structure
+
+    Raises:
+        TypeError: if ``model`` contains an object of unsupported type
+
+    Returns:
+        Union[torch.Tensor, list, Any], array: first elements of flat arranged into the nested
+        structure of model, unused elements of flat
+    """
+    if isinstance(model, (numbers.Number, str)):
+        return flat[0], flat[1:]
+
+    if isinstance(model, torch.Tensor):
+        idx = model.size
```

Comment:
```suggestion
        idx = model.numel()
```
If I understand from the other version of `_unflatten` we have.

```python
x = torch.tensor(np.random.random((3, 3, 3)))
print(x.size)
print(x.size())
print(x.numel())
```